server/task/src/main/java/org/apache/james/task/Task.java

```diff
@@ -75,4 +76,18 @@ static Result combine(Result result1, Result result2) {
      */
     Result run();
 
+
+    default String type() {
+        return UNKNOWN;
+    }
+
+    default Optional<Object> details() {
```

Comment:
we cannot have better than Object? Maybe an interface at least?

Reply:
Sold for an interface!